x-pack/plugins/maps/public/classes/styles/vector/vector_style.tsx

```diff
@@ -239,11 +241,173 @@ export class VectorStyle implements IVectorStyle {
     );
   }
 
+  async _updateFieldsInDescriptor(
+    nextFields: IField[],
+    styleFieldsHelper: StyleFieldsHelper,
+    originalProperties: VectorStylePropertiesDescriptor,
+    previousFields: IField[]
+  ) {
+    const invalidStyleNames: VECTOR_STYLES[] = (Object.keys(
+      originalProperties
+    ) as VECTOR_STYLES[]).filter((key) => {
+      const dynamicOptions = getDynamicOptions(originalProperties, key);
+      if (!dynamicOptions || !dynamicOptions.field || !dynamicOptions.field.name) {
+        return;
+      }
+
+      const hasMatchingField = nextFields.some((field) => {
+        return (
+          dynamicOptions && dynamicOptions.field && dynamicOptions.field.name === field.getName()
+        );
+      });
+      return !hasMatchingField;
+    });
+
+    let hasChanges = false;
+
+    invalidStyleNames.forEach((invalidStyleName) => {
+      for (let i = 0; i < previousFields.length; i++) {
+        const previousField = previousFields[i];
+        const nextField = nextFields[i];
+        if (previousField.isEqual(nextField)) {
+          continue;
+        }
+        const isFieldDataTypeCompatible = styleFieldsHelper.hasFieldForStyle(
+          nextField,
+          invalidStyleName
+        );
+        if (!isFieldDataTypeCompatible) {
+          return;
+        }
+        const newFieldDescriptor = rectifyFieldDescriptor(nextField as IESAggField, {
+          origin: previousField.getOrigin(),
+          name: previousField.getName(),
+        });
+        hasChanges = true;
+        (originalProperties[invalidStyleName]!
+          .options! as DynamicStylePropertyOptions).field = newFieldDescriptor;
+      }
+    });
+
+    return {
+      hasChanges,
+      nextStyleDescriptor: VectorStyle.createDescriptor(originalProperties, this.isTimeAware()),
+    };
+  }
+
+  async _deleteFieldsFromDescriptorAndUpdateStyling(
+    nextFields: IField[],
+    styleFieldsHelper: StyleFieldsHelper,
+    originalProperties: VectorStylePropertiesDescriptor,
```

Comment:
Why is `originalProperties` passed in a parameter when it can just be retrieved from the instance? So instead of passing in, just `const originalProperties = this.getRawProperties();` directly in the method.